### DISCUSSION TITLE
How do I translate a simple server with graceful shutdown to axum 0.7?

### DISCUSSION THREAD

**Discussion by ISibboI** [tokio-rs/axum#2403] [Q&A] [answered]
### Summary

Former code (axum 0.6):

```
let router = Router::new(); // configuration omitted

axum::Server::bind(SocketAddr::from(([0, 0, 0, 0], 8093)))
        .serve(router.into_make_service())
        .with_graceful_shutdown(shutdown_signal()) // implementation of shutdown_signal omitted
        .await
        .unwrap();
```

How does this look like in axum 0.7? Without the graceful shutdown, I understand that this would simply translate using the new `axum::serve` method. But using that method, I don't know how to get graceful shutdown.

I saw an example that does exactly this, but it introduces 100 lines or so of boilerplate: https://github.com/tokio-rs/axum/blob/main/examples/graceful-shutdown/src/main.rs

Is this possible without all the boilerplate, and instead with just a simple method call, as it was before?

### axum version

0.7.1

**Comment by davidpdrsn** [comment, marked as answer]
Please see the comment at the top of the file https://github.com/tokio-rs/axum/blob/main/examples/graceful-shutdown/src/main.rs#L8-L9.

If there was an easier way to do it then that's what we'd use in the example. Unfortunately there is not but we're working on adding it https://github.com/tokio-rs/axum/pull/2398. Its currently blocked by hyper.

**Reply by ISibboI** [reply]
Thanks!
